js/components/input-number.tsx

```diff
@@ -91,22 +103,31 @@ class InputNumber extends React.Component<IInputProps, IInputNumberState> {
             if (!manywho.utils.isNullOrUndefined(model.attributes.step))
                 step = model.attributes.step;
         }
+        const props: any = {};
+        if (this.inputRef !== null) {
+            props.ref = this.inputRef;
+        }
 
-        return <input id={this.props.id}
-            value={this.state.value}
-            placeholder={this.props.placeholder}
-            className="form-control"
-            type="number"
-            style={style}
-            max={max}
-            min={min}
-            step={step}
-            readOnly={this.props.readOnly}
-            disabled={this.props.disabled}
-            required={this.props.required}
-            onChange={!this.props.isDesignTime && this.onChange}
-            onBlur={this.props.onBlur}
-            autoComplete={this.props.autocomplete} />;
+        return (
+            <input
+                id={this.props.id}
+                value={this.state.value}
+                placeholder={this.props.placeholder}
+                className="form-control"
+                type="number"
+                style={style}
+                max={max}
+                min={min}
+                step={step}
+                readOnly={this.props.readOnly}
+                disabled={this.props.disabled}
+                required={this.props.required}
+                onChange={!this.props.isDesignTime && this.onChange}
+                onBlur={this.props.onBlur}
+                autoComplete={this.props.autocomplete}
+                {...props}
```

Comment:
Is there any harm in just always defining a ref for this component? I think then there would be no need use the spread operator here and just pass the ref in as a single prop.
Means you could get rid of conditional logic on lines 22 and 106, unless I'm missing something.

Reply:
In the react docs it says that references shouldn't be overused, In this case will be used just in the specific case that autofocus is needed and the input is an autofocus candidate, but not by default. In most of the scenarios will not be a reference linked with this element. 
I think it could be possible to get rid of the reference in the future.